overlord/state/taskrunner.go

```diff
@@ -274,6 +301,22 @@ func (r *TaskRunner) run(t *Task) {
 			var next []*Task
 			switch t.Status() {
 			case DoingStatus:
+				unblocked, minCheckAfter, err := r.checkBlockingConditions(t)
```

Comment:
my preference would be to make work more like how we make at-time work, with logic in the main run loop

Reply:
I mean the new extra logic to block a task should hang mainly around:  https://github.com/canonical/snapd/blob/master/overlord/state/taskrunner.go#L453-L469

Reply:
So the main checks that prevents us from running blocked tasks is in Ensure. This bit here basically checks for blockers after we run to check if the just completed task run has added any blockers and if so, disregards the run. The issue with moving that into the Ensure loop is that we're doing that check in another loop run and meanwhile the task is set as Done which would cause its dependencies to run (if they were sorted before the blocked task in `state.tasks`